controllers/application/configmap.go

```diff
@@ -99,3 +127,56 @@ func (r *ApplicationReconciler) setupGCPAuthConfigMap(ctx context.Context, gcpId
 
 	return err
 }
+
+func (r *ApplicationReconciler) setupSkiperatorConfigMap(ctx context.Context, skiperatorConfigMap corev1.ConfigMap, application *skiperatorv1alpha1.Application) error {
+	err := ctrlutil.SetControllerReference(application, &skiperatorConfigMap, r.GetScheme())
+	if err != nil {
+		r.SetControllerError(ctx, application, controllerName, err)
+		return err
+	}
+	r.SetLabelsFromApplication(ctx, &skiperatorConfigMap, *application)
+
+	mapData := skiperatorConfigMap.Data
+	if len(mapData) == 0 {
+		mapData = make(map[string]string)
+	}
+
+	// // We only want to set the istio CPU Request if it is not already set
+	// _, present := mapData[istioSidecarAnnotation]
+	// if !present {
+	// mapData[istioSidecarAnnotation] = getDefaultIstioCPURequestFromEnv(r.Environment)
+	// }
+	// ^ Do we only want to do this if its not already set? Should we allow overriding the default request per application?
```

Comment:
Would like some comments here. You cannot edit the ConfigMap now.

(Well you can, and it wont be reconciled, but as soon as a deployment is reconciled, so is the configmap and your changes will not matter)

Reply:
A common pattern is to add a suffix to each revision of the configmap to trigger a redeployment. E.g.

```
foo-skiperator-config-ffaa00
```

where `ffaa00` is a hash over the ConfigMap's contents. This specific name is in turn referenced in the `skiperator` deployment. When content in the ConfigMap changes, a new name is generated and the deployment is updated to use the new ConfigMap.

---

I'm not sure if that's the best option here or not – let's discuss offline